lib/nx_zlib.c

```diff
@@ -761,11 +763,26 @@ static int nx_enumerate_engines()
 
 	d = opendir(DEVICE_TREE);
 	if (d == NULL){
-		prt_err("open device tree dir failed.\n");
+		prt_info("open device tree dir failed.\n");
+
+		/* In a restricted container environment, the device tree
+		   directory may not be available. We can still find out if
+		   this is a PowerVM environment.  */
+		if (access(LPARCFG_FILE, F_OK) == 0) {
+			nx_config.virtualization = POWERVM;
+			return 1;
```

Comment:
@tuliom What if the code runs in a PowerVM system which doesn't have `/dev/crypto/nx-gzip` configured? Isn't it better to just set the virtualization type here and leave the number of engines to be decided by the next check?

Reply:
@mscastanho I was trying to avoid this because I didn't want to conflict with the code in `open_device_nodes()` which properly check for the existence, permission and being able to open this device.
Notice the current code for PowerVM returns 1 device without checking if `/dev/crypto/nx-gzip` exists. It seems to return that even on Power9, which is not supported.

If we change this code in order to check for at least existence and permission, should we return 0 if any of them fail?

Reply:
Ok, if the previous code was already returning 1 I think it's better to leave it as-is for now. We know things are working, let's not mess up this late in the game. :shipit: